### ISSUE TITLE
Replace $cookies global and related logic

### ISSUE THREAD

**Issue by cnotv** [rancher/dashboard#11392]
# Description

From a removal attempt (https://github.com/rancher/dashboard/issues/11272), the global $cookies and 2-way binding logic adopted are defined from the plugin `cookie-universal-nuxt`. This has been replaced with a more generic one `cookie-universal` https://github.com/rancher/dashboard/issues/11321
As the logic creates several issues, replace with some more conventional patterns.
The main reason is the presence of too many states: cookies, global, and Vuex.

Occurrences are not so high:
https://github.com/search?q=%24cookies+repo%3Arancher%2Fdashboard&type=code&ref=advsearch

<img width="337" alt="Screenshot 2024-07-01 at 17 26 43" src="https://github.com/rancher/dashboard/assets/5009481/cdbb59b3-46c4-4a7e-9486-fe56b07b93ab">

**Comment by nwmac** [comment]
@cnotv Can we push this to 2.11?

**Comment by cnotv** [comment]
> @cnotv Can we push this to 2.11?

Yes definitely. @rak-phillip has found a generic replacement that makes this change not strictly necessary. unless we want to change the logic around.

**Comment by codyrancher** [comment]
@cnotv are you wanting to remove the use of cookies all together or are you just wanting to remove the $cookies function that gets injected into the vue instance?

**Comment by cnotv** [comment]
Hard exercise for my memory. If I am not wrong, the idea was initially to remove everything about Nuxt. Then we crossed some use cases of the cookies as well. My suggestion was to remove the global.

Ideally you want to deal just with the Vuex state and the Cookies be mapped to it so you can reference (and debug) always the store.

**Comment by codyrancher** [comment]
We should verify some auth providers: Amazon cognito, keycloak, okta saml after an issue was noticed.

**Comment by aalves08** [comment]
Validating https://github.com/rancher/dashboard/issues/13684 should also cover all the cases for this issue

**Comment by yonasberhe23** [comment]
Testing covered in https://github.com/rancher/dashboard/issues/13684 and during pr review process https://github.com/rancher/dashboard/pull/15515#pullrequestreview-3264955470
